### ISSUE TITLE
Bug: Problem identifying correct release with multiple different releases

### ISSUE THREAD

**Issue by drzero42** [marcosnils/bin#99]
Scenario:
```
abo@ernest:~$ bin install github.com/kubernetes-sigs/kustomize
   • Getting latest release for kubernetes-sigs/kustomize
   ⨯ command failed            error=Could not find any compatible files
abo@ernest:~$ bin install https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.1.2
   • Getting kustomize/v4.1.2 release for kubernetes-sigs/kustomize
   • Starting download of https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v4.1.2/kustomize_v4.1.2_linux_amd64.tar.gz
4.97 MiB / 4.97 MiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 5.74 MiB p/s 1s
   • Copying for kustomize@kustomize/v4.1.2 into /home/abo/.local/bin/kustomize
   • Done installing kustomize kustomize/v4.1.2
abo@ernest:~$ bin update kustomize
   • /home/abo/.local/bin/kustomize kustomize/v4.1.2 -> kyaml/v0.10.19 (https://github.com/kubernetes-sigs/kustomize/releases/tag/kyaml/v0.10.19)
```

So `bin` runs into problems when a repo releases multiple different things, which is fair I guess.
Anything we can do to make it possible to install releases from such repos?

**Comment by marcosnils** [comment]
> Anything we can do to make it possible to install releases from such repos?

This is going to be very tricky.. since if repo use any convention that they want it's going to be very difficult for us to be able to do something about it. I believe not even github thought about this use-case of one repository having multiple releases of multiple projects. 

I saw this in the past but I've decided not to do so anything about it since it's the only case that I came across so far. If you have any `simple` way of fixing this I'm happy to hear options, but otherwise, I don't even think it's worth investing the time to fix this only case.

**Comment by drzero42** [comment]
My first thought was that since the repo uses tagging that contains names of the releases (ie. `kustomize` and `kyaml`) we could allow setting a prefix for tags. That unfortunately probably won't work as the "latest release" only points to whatever was released last, switching between different products, so there won't always be a `kustomize` release marked as "latest release".
Can `bin` figure out the latest version aside from what is marked as "latest release"?

**Comment by marcosnils** [comment]
> Can `bin` figure out the latest version aside from what is marked as "latest release"?

Yes, we can do this since the GH releases API returns the releases ordered by date. The "problem" with this is that we'll have to supply some sort of prefix to `bin install / update` to let it know from which release we want to pick the binaries from. This is the part that I don't like for the following reasons:

1. If we install things with `bin install --prefix kustomize/ <repo_url>` that means that bin will have to save the prefix in the config and use it on every update. If this is the case and the repo decides to use a different release name for whatever reason, then `bin update` will just break :(

2. Based on 1, if we decide not to persist the prefix, then the UX around `bin update` will be completely broken as there will be some binaries which can't be updated.

I'd like to provide a "nice" alternative for this, but so far the only case where this has happened is the kustomize repo. Not sure if it makes sense to invest the time in fixing ATM.

**Comment by danielhoherd** [comment]
I am running into this same problem with https://github.com/vectordotdev/vector where there are packaged releases for `vdev` and `vector`. The solution that I expected to find was a substring or regex match that could be configured in `config.json:.bin/[tool_path].something`.

**Comment by marcosnils** [comment]
@danielhoherd this has been implemented in https://github.com/marcosnils/bin/commit/77e43dcbc08e7bb1961730a8b27eb560f8302e5f
https://github.com/marcosnils/bin/issues/279#issuecomment-4241578275

forgot to add some docs about it. Mind checking if this fixes your issue?

**Comment by danielhoherd** [comment]
~@marcosnils it did not work. the filter appears to be unanchored:~

```
$ bin install 'https://github.com/vectordotdev/vector?filter=v0.*'
  • Getting latest release for vectordotdev/vector
  • Starting download of https://api.github.com/repos/vectordotdev/vector/releases/assets/418865951
3.10 MiB / 3.10 MiB [-----------------------------------------] 100.00% ? p/s 0s
  • Copying for vdev@vdev-v0.3.3 into /Users/danielh/bin/vdev
  • Done installing vdev vdev-v0.3.3
```

~`?filter=v0.*` appears to be matching on `vdev-v0.3.3`~

**EDIT**: my version of `bin` was not latest. This does work with the latest version.

**Comment by marcosnils** [comment]
this works for what you're trying to achieve: bin install --debug
https://github.com/vectordotdev/vector?filter=v0*

I'm trying to find if there's a better pattern instead of using `v0` given
that after v1 is released, this will stop working

**Comment by danielhoherd** [comment]
~@marcosnils that does not work for me:~
```
$ bin install --debug https://github.com/vectordotdev/vector?filter=v0*
  • debug logs enabled, version: 0.24.2
commit: 7e93aa50b7238155f3c9bcac786fdee1be292a49
built at: 2025-11-06T19:09:05Z
built by: goreleaser

  • Config directory is: /Users/danielh/.config/bin
  • Download path set to /Users/danielh/bin
  • Using provider 'github' for 'https://github.com/vectordotdev/vector?filter=v0*'
  • Getting latest release for vectordotdev/vector
  • Candidate vdev-aarch64-apple-darwin-v0.3.3.tgz contains darwin. Adding score 10
  • Keeping vdev-aarch64-apple-darwin-v0.3.3.tgz (URL https://api.github.com/repos/vectordotdev/vector/releases/assets/418865951) with highest score 10
  • Checking binary from https://api.github.com/repos/vectordotdev/vector/releases/assets/418865951
  • Starting download of https://api.github.com/repos/vectordotdev/vector/releases/assets/418865951
3.10 MiB / 3.10 MiB [-------------------------------------------------------------------------------------------------------------------------] 100.00% ? p/s 0s
  • Copying for vdev@vdev-v0.3.3 into /Users/danielh/bin/vdev
  • Done installing vdev vdev-v0.3.3
$ bin --version
bin version 0.24.2
commit: 7e93aa50b7238155f3c9bcac786fdee1be292a49
built at: 2025-11-06T19:09:05Z
built by: goreleaser
```

~(I also tried it with quotes around the URL, same result.)~

One of the challenges with this repo is that the `vector` releases do not have a unique positive matchable substring, it is just `vX.Y.Z`, whereas the thing I want to exclude does have a positive match `vdev-vX.Y.Z`

**EDIT**: I saw that my version of `bin` is old. I updated it and the above does work.

**Comment by marcosnils** [comment]
Hi Daniel,

You're running an old version of bin. Upgrading should make this work.
